images/dynamodb_local/src/image.rs

```diff
@@ -0,0 +1,85 @@
+use std::{env::var, thread::sleep, time::Duration};
+use tc_core::{Image, Container, Docker, WaitForMessage};
+
+const ADDITIONAL_SLEEP_PERIOD: &'static str = "DYNAMODB_ADDITIONAL_SLEEP_PERIOD";
+const DEFAULT_WAIT: u64 = 2000;
+const CONTAINER_IDENTIFIER: &'static str = "amazon/dynamodb-local";
+const DEFAULT_TAG: &'static str = "latest";
+
+
+#[derive(Debug, Default, Clone)]
+pub struct DynamoDbArgs;
+
+impl IntoIterator for DynamoDbArgs {
+    type Item = String;
+    type IntoIter = ::std::vec::IntoIter<String>;
+
+    fn into_iter(self) -> <Self as IntoIterator>::IntoIter {
+        vec![].into_iter()
+    }
+}
+
+#[derive(Debug)]
+pub struct DynamoDb {
+    tag: String,
+    arguments: DynamoDbArgs
+}
+
+impl Default for DynamoDb {
+    fn default() -> Self {
+        DynamoDb {
+            tag: DEFAULT_TAG.to_string(),
+            arguments: DynamoDbArgs {}
+        }
+    }
+}
+
+impl Image for DynamoDb {
+    type Args = DynamoDbArgs;
+
+    fn descriptor(&self) -> String {
+        format!("{}:{}", CONTAINER_IDENTIFIER, &self.tag)
+    }
+
+    fn wait_until_ready<D: Docker>(&self, container: &Container<D, Self>) {
+        container
+            .logs()
+            .stdout
+            .wait_for_message("Initializing DynamoDB Local with the following configuration")
+            .unwrap();
+
+        let additional_sleep_period = var(ADDITIONAL_SLEEP_PERIOD)
```

Comment:
I see you got inspired by the `bitcoind` image :)
Is the additional sleep period actually necessary for this image or is the container ready after the above log message?
If yes, please remove this part of the code, we try to have as minimal configuration as possible.

Reply:
I would like to move away from the sleep period completely at some point. I was too lazy to add something like check for the port to be listening. When Dynamo starts up it will output the following:
`Initializing DynamoDB Local with the following configuration:
Port:	8000
InMemory:	true
DbPath:	null
SharedDb:	false
shouldDelayTransientStatuses:	false
CorsParams:	*`

The only static line is the first -- which isn't really that ideal. 

Originally I set the sleep time to 1 sec. Which seemed adequate most the time, but would occasionally fail. I increased it to 2 seconds, and it doesn't seem to have an issue on my brand new MBP. My fear is that I if I remove any ability override, someone else's hardware isn't going perform the same and it will always fear. So long term, I should probably just come  up with a better solution.

Reply:
Nah, things should be simple. No need for port checking :)
One interesting thing to mention is that the code behind `wait_for_message` uses `contains` so you can match on partial lines if the whole line is not static.

I see your point with the sleep period. I had to add it to the bitcoind image for the exact same shitty reason. It's a mess :D

Maybe we can build an abstraction at some point to unify that.